src/session.rs

```diff
@@ -36,12 +44,48 @@ impl Session {
             .expect("Driver should not panic")?;
         return Ok(RequestResult { inner: result });
     }
-}
 
-#[pyclass]
-#[derive(Clone)]
-pub(crate) struct PyPreparedStatement {
-    pub(crate) _inner: Arc<scylla::statement::prepared::PreparedStatement>,
+    async fn prepare(&self, request: Py<PyString>) -> PyResult<PyPreparedStatement> {
+        let request_string = Python::with_gil(|py| request.to_str(py))?.to_string();
+        let session_clone = Arc::clone(&self._inner);
+        let result = RUNTIME
+            .spawn(async move {
+                session_clone.prepare(request_string).await.map_err(|e| {
+                    PyRuntimeError::new_err(format!("Failed to deserialize metadata: {}", e))
+                })
+            })
+            .await
+            .expect("Driver should not panic")?;
+        return Ok(PyPreparedStatement {
+            _inner: (Arc::new(result)),
+        });
```

Comment:
:question: What for do we spawn the task on tokio runtime? Can't we await it specifically on the calling thread?

Reply:
I invite you to checkout the PR locally to try and do this. It will be a good exercise to better understand limitations of async ffi.

Reply:
In Node.js Driver, the binding library hides task spawning, so I don't even know.
But you're right, this limitation makes perfect sense: Python threads most probably cannot be tied to Rust reactor, because that would conflict with Python reactor.